packages/announcement-controller/src/AnnouncementController.ts

```diff
@@ -1,96 +1,116 @@
-  BaseState,
+  BaseControllerV2,
+  RestrictedControllerMessenger,
 } from '@metamask/base-controller';
 
-interface ViewedAnnouncement {
+type ViewedAnnouncement = {
   [id: number]: boolean;
-}
+};
 
-interface Announcement {
+type Announcement = {
   id: number;
   date: string;
-}
-
-interface StateAnnouncement extends Announcement {
-  isShown: boolean;
-}
+};
 
-/**
- * A map of announcement ids to Announcement objects
- */
-interface AnnouncementMap {
-  [id: number]: Announcement;
-}
+type StateAnnouncement = Announcement & { isShown: boolean };
 
 /**
  * A map of announcement ids to StateAnnouncement objects
  */
-export interface StateAnnouncementMap {
+export type StateAnnouncementMap = {
   [id: number]: StateAnnouncement;
-}
-
-/**
- * AnnouncementConfig will hold the active announcements
- */
-export interface AnnouncementConfig extends BaseConfig {
-  allAnnouncements: AnnouncementMap;
-}
+};
 
 /**
  * Announcement state will hold all the seen and unseen announcements
  * that are still active
  */
-export interface AnnouncementState extends BaseState {
+export type AnnouncementState = {
   announcements: StateAnnouncementMap;
-}
+};
+
+export type AnnouncementControllerActions = GetAnnouncementState;
+export type AnnouncementControllerEvents = AnnouncementStateChange;
+
+export type GetAnnouncementState = {
+  type: `${typeof controllerName}:getState`;
+  handler: () => AnnouncementState;
+};
+
+export type AnnouncementStateChange = {
+  type: `${typeof controllerName}:stateChange`;
+  payload: [AnnouncementState, Patch[]];
+};
+
+const controllerName = 'AnnouncementController';
 
 const defaultState = {
   announcements: {},
 };
 
+const metadata = {
+  announcements: {
+    persist: true,
+    anonymous: false,
+  },
+};
+
+export type AnnouncementControllerMessenger = RestrictedControllerMessenger<
+  typeof controllerName,
+  AnnouncementControllerActions,
+  AnnouncementControllerEvents,
+  never,
+  never
+>;
+
 /**
  * Controller for managing in-app announcements.
  */
-export class AnnouncementController extends BaseController<
-  AnnouncementConfig,
-  AnnouncementState
+export class AnnouncementController extends BaseControllerV2<
+  typeof controllerName,
+  AnnouncementState,
+  AnnouncementControllerMessenger
 > {
   /**
    * Creates a AnnouncementController instance.
    *
-   * @param config - Initial options used to configure this controller.
-   * @param state - Initial state to set on this controller.
+   * @param args - The arguments to this function.
+   * @param args.messenger - Messenger used to communicate with BaseV2 controller.
+   * @param args.state - Initial state to set on this controller.
    */
-  constructor(config: AnnouncementConfig, state?: AnnouncementState) {
-    super(config, state || defaultState);
-    this.initialize();
-    this._addAnnouncements();
+  constructor({
+    messenger,
+    state,
+  }: {
+    messenger: AnnouncementControllerMessenger;
+    state?: AnnouncementState;
```

Comment:
Also, it looks like it is no longer possible to pass a set of announcements via `allAnnouncements`. The BaseController v1 way of allowing for on-the-fly configuration was via the constructor's `config` argument. So in this case, you could say:

``` typescript
new AnnouncementController(
  // config
  {
    allAnnouncements: { ... }
  },
  // state
  {
    announcements: { ... }
  }
)
```

Since the BaseControllerV2 now takes only one argument instead of two and since that argument is an object, the way to do this now is to allow that object to have more properties than just `messenger` and `state`. For instance, the [options that PermissionController takes](https://github.com/MetaMask/controllers/blob/c419a244072655a246a0a40ba2fad30c0db58f61/packages/permission-controller/src/PermissionController.ts#L444) still includes `messenger` and `state` but also several other options.

So in this case, I think we would want to have this constructor take `allAnnouncements` and then pass that to `this._addAnnouncements` instead of grabbing the announcements from `state`. After all, if `this._addAnnouncements` adds announcements to state, but we've already initialized the state on line 90, then it's not really doing anything.